### ISSUE TITLE
Fix another unstable test after eager rent

### ISSUE THREAD

**Issue by mvilnes** [solana-labs-community/solana#109]
#### Problem
`test_bank_get_program_accounts` is flaky since eager rent was implemented.
```
thread 'bank::tests::test_bank_get_program_accounts' panicked at 'assertion failed: `(left == right)`
left: `[(2ENdwbDcvnewdQZwY4mAJ9sApwCMBo3AmXoQnUzG3m8p, Account { lamports: 1 data.len: 0 owner: 8qbHbw2BbbTHBW1sbeqakYXVKRQM8Ne7pLK7m6CVfeR executable: false rent_epoch: 0 })]`,
right: `[]`', runtime/src/bank.rs:5868:9
```

#### Summary of Changes
Preserve lazy rent for this test, seems to resolve the flakiness (about 40 local runs succeeded, compared with failures every 3-6).

However, @CrieasofCarrots, I would like your insight as to whether (a) it is correct that this account should show that it was modified since parent, even though the balance hasn't changes; (b) this is the right solution.
Turning off rent seems like another option, since this test is meant to detect non-rent account changes. (It does fix the flakiness)

I'm actually a bit surprised there aren't more flaky tests like this, but I guess there aren't any other tests using a chain of banks and expecting accounts to stay unchanged in this way. I did a quick scan through bank.rs and didn't see any.

Fyi @jeffweshington

**Comment by CrieasofCarrots** [comment]
Thanks for spotting on this!

Well, there was and is some.... ;) (Past: #10031, waiting for my love: https://buildkite.com/solana-labs/solana/builds/24442#d1c0a942-4d7b-491e-b6b6-71653ffeab06/159-287). Sorry for ongoing trouble.

I'll work on these once devnet bloat snapshot issue is settled down asap.

**Comment by CrieasofCarrots** [comment]
Yeah, it's correct. undocumented (#10133) and unintuitive. :( But eager rent collection always updates account even if the balance hasn't changed. That's intended behavior to purge too old account states for each epoch #8931 .

By the way, I've noticed all touched accounts by eager rent collection would be exposed to get_program_accounts_modified_since_parent and the like, eventually RPC. Would that be problem? I mean, those RPCs will start to return those accounts as noise.

Yeah, that flag is exactly designed for that. Once eager rent collection is enabled across the board, I'll rewrite those tests.

**Comment by jeffweshington** [comment]
LGTM! Really thanks for working on my bug..!

**Comment by t-neIson** [comment]
Yes, I do think it is a slight UX problem, as it will be confusing for users for accounts to appear as `modified_since_parent` but show no changes. We might want to investigate a way to prevent that. Or I suppose we could toggle a bit in the account to indicate that rent was checked, so there is actually a change in the account state. I'll create an issue for it.
